astropy/modeling/tests/test_separable.py

```diff
@@ -52,7 +52,14 @@
     'cm7': (map2 | p2 & sh1,
             (np.array([False, True]),
              np.array([[True, False], [False, True]]))
-            )
+            ),
+    'cm8': (rot & (sh1 & sh2),
+            (np.array([False, False, True, True]),
+             np.array([[True, True, False, False],
+                       [True, True, False, False],
+                       [False, False, True, False],
+                       [False, False, False, True]]))
+            ),
```

Comment:
Since was a previously undetected problem could we add a few more test cases for other model combinations. Such as
```
rot & sh1 & sh2
(rot & sh1) & sh2
```

Reply:
Feel free to throw in a few more pathological test cases too, I would like to more thoroughly test this.

Reply:
Added a few more, happy to add more if you can think of them but my creativity failed me.